examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,504 @@
+            llm_config (OpenAIGPTConfig): The LLM configuration for the agent.
+            system_message (str): The system message to guide the agent's LLM.
+
+        Returns:
+            ChatAgent: A configured ChatAgent instance.
+        """
+    return ChatAgent(
+        ChatAgentConfig(
+            llm=llm_config,
+            name=name,
+            system_message=system_message,
+        )
+    )
+
+
+def run_debate() -> None:
+    """Execute the main debate logic.
+
+    Orchestrates the debate process, including setup, user input, LLM agent
+    interactions, and final feedback. Handles both user-guided and LLM-
+    delegated debates.
+
+    This function:
+    1. Loads global settings and the base LLM configurations.
+    2. Prompts the user to confirm if they want to use same LLM for all agents.
+    3. Prompts the user to select a debate topic and a side(Pro or Con).
+    4. Sets up pro, con, and feedback agents.
+    5. Runs the debate for a specified number of turns, either interactively
+       or autonomously.
+    6. Provides a feedback summary at the end.
+    """
+
+    global_settings = get_global_settings(nocache=True)
+    lr.utils.configuration.set_global(global_settings)
+
+    same_llm: bool = is_same_llm_for_all_agents()
+    llm_delegate: bool = is_llm_delegate()
+    metaphor_search: bool = is_metaphor_search_key_set()
+
+    # Get base LLM configuration
+    if same_llm:
+
+        shared_agent_config: OpenAIGPTConfig = get_base_llm_config(select_model("main LLM"))
+        pro_agent_config = con_agent_config = shared_agent_config
+
+        # Create feedback_agent_config by modifying the temperature of shared_agent_config
+        feedback_agent_config: OpenAIGPTConfig = OpenAIGPTConfig(
+            chat_model=shared_agent_config.chat_model,
+            min_output_tokens=shared_agent_config.min_output_tokens,
+            max_output_tokens=shared_agent_config.max_output_tokens,
+            temperature=0.2,  # Override temperature
+            seed=shared_agent_config.seed,
+        )
+    else:
+        pro_agent_config: OpenAIGPTConfig = get_base_llm_config(select_model("for Pro Agent"))
+        con_agent_config: OpenAIGPTConfig = get_base_llm_config(select_model("for Con Agent"))
+        feedback_agent_config: OpenAIGPTConfig = get_base_llm_config(select_model("feedback"), temperature=0.2)
+
+    system_messages: SystemMessages = load_system_messages("examples/multi-agent-debate/system_messages.json")
+    topic_name, pro_key, con_key, side = select_topic_and_setup_side(system_messages)
+
+    # Prompt for number of debate turns
+    max_turns: int = int(
+        Prompt.ask(f"How many turns should the debate continue for? (Default is {DEFAULT_TURN_COUNT})",
+                   default=DEFAULT_TURN_COUNT)
+    )
+
+    # Generate the system message
+    metaphor_search_agent_system_message = generate_metaphor_search_agent_system_message(system_messages, pro_key,
```

Comment:
avoid long vbl and fn names. One suggestion `x_agent` -> `x_task` instead of `x_agent_task`.

Reply:
done